runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -2565,6 +2583,23 @@ private void doFetchInitialBegin(
                         .sizeof()));
         }
 
+        private void doFetchInitialFlush(
+            long traceId,
+            long partitionOffset)
+        {
+            KafkaFlushExFW kafkaFlushEx = kafkaFlushExRW.wrap(extBuffer, 0, extBuffer.capacity())
+                .typeId(kafkaTypeId)
+                .fetch(f -> f
+                    .partition(p -> p.partitionId(partitionId)
```

Comment:
I believe this might not be 100% correct. While doing the end-to-end testing I discovered a weird indeterministic behaviour. Scenario:

- Mqtt flush comes in (adding a new subscription, with send retained true)
- send the kafka flush to the messages stream (this is what's triggering this fetch flush)
- open the retained stream with the undelivered retained filters
- deliver retained messages over the retained stream
- close the retained stream
- publish a new data, that arrives in the messages merged stream

After this the retained message that was already delivered over the retained stream gets delivered again (not always) in the MESSAGES stream. I feel like this might have to do with setting the partition offsets when doing the flush, but couldn't figure out yet.

Reply:
We are not supposed to set the partition offsets during the `FLUSH` because we want the cursor offset position to remain unchanged, just pick up from where it is with an adjusted filter, agree?

Reply:
You're right. I did the change